oeps/oep-0011.rst

```diff
@@ -0,0 +1,412 @@
+   plans are being made to evaluate its usage. See `React at edX`_ for
+   more details.
+
+2. **How should offline support be provided?**
+
+   For many mobile users, it would be very useful if they could continue
+   their Open edX courses while offline. This is true both for true mobile
+   apps as well as for responsive web applications. This is an area that
+   has not been explored much up to now, but which will only become more
+   important going forward. In particular, we should look at the great
+   capabilities provided by service workers. See the article
+   `Service Workers: An Introduction`_ for a good starting point.
+
+3. **Should Universal JavaScript be considered?**
+
+   There are many companies that are seeing performance benefits by
+   pre-rendering their dynamic client-side applications on the server first.
+   The most common approach is to run the same JavaScript code both in the
+   browser and using Node on the server. This requires the use of a JavaScript
+   framework that does not perform direct manipulation of the DOM, but instead
+   works with an abstraction of the desired HTML. For example, Netflix is
+   seeing great performance gains by using `React`_ in this way, as can be
+   seen in the blog post `Making Netflix Faster`_.
+
+4. **Where do web components fit with Open edX applications?**
+
+   Web Components is a W3C recommendation that allows developers to create
+   custom HTML elements for use within their web applications. The
+   `Polymer Project` provides a library that allows web components to be
+   built in most modern browsers. It seems that there is a useful overlap
+   between web components and Open edX's `XBlocks`_
+
+5. **Should HTTP/2 be used in preference to asset bundling?**
+
+   The Hypertext Transfer Protocol Version 2 (HTTP/2) enables more efficient
+   network communication between clients and servers. In particular, it
+   allows for multiple requests to be handled with the same TCP connection,
+   which removes most of the benefits of asset bundling. It is not yet
+   clear whether this is ready for full scale adoption, but it would reduce
+   a lot of complexity from application's asset pipelines.
+
+6. **Should Jinja2 be considered in preference to Django Templates?**
+
+   `Jinja2`_ is a popular alternative to Django Templates, that has not been
+   evaluated for use within Open edX.
+
+
+Rejected Alternatives
+=====================
+
+Note: this section needs fleshing out...
+
+.. _Reasons for rejecting Angular:
+1. **Angular**
+
+   Needs fleshing out...
+
+.. _Reasons for rejecting Bootstrap:
+2. **Bootstrap**
+
+   An evaluation was performed,
+ - size
+ - not accessible (although there is this an accessibility plugin)
+ - too inflexible
+ - might want to consider compatibility to allow mixing Bootstrap components
+   into pattern library pages. See the edX Bootstrap theme provided by
+   the community as one approach.
+
+   Needs fleshing out...
```

Comment:
I can't speak to the reasons we chose not to use Bootstrap because I wasn't around when we made that decision initially, but to me one of the biggest things against sites built with Bootstrap is they feel cheap and routine. Bootstrap feels to me like a house in a development where everything looks nice and functionally *works*, but you look next door and across the street and you realize all of the houses are the exact same. Maybe their driveway is on the other side of the house from the one next door and they have different shingles and some nice flowers in the front yard, but you can tell walking past that they all have basically the same floorplan, and they all feel cheap and interchangeable.

Accordingly, most Bootstrap sites I see are either hackathon projects or ProductHunted landing pages for MVP Uber-for-dogs startups. I'm not aware of any really serious sites with good branding/UX that outsource that work to Bootstrap. Developing our own style and components and building blocks is a lot more work but it's really important in showing who we are.